code/fighter.gd

```diff
@@ -95,33 +102,52 @@ func _fixed_process(dt):
 				FreezeInput = Vector2(0, 1)
 				FreezeInputRecorded = true
 	
-func shoot():
+func shoot(BulletType):
 	var GameNode = get_tree().get_root().get_node("Game")
 	var BulletNode = BulletScene.instance()
 	BulletNode.set_pos(get_pos())
-	BulletNode.set_velocity(BulletSpeed * BulletDir)
+	BulletNode.set_name("Bullet")
+
+	var BulletVelocity
+	var BulletPower
+	# NOTE(hugo) : There are still no switch statement in Godot for now :(
+	if(BulletType == BULLET_TYPE_FAST):
+		BulletVelocity = BulletSpeed * BulletDir
+		BulletPower = 10
+
+	elif(BulletType == BULLET_TYPE_STRONG):
+		BulletVelocity = 0.5 * BulletSpeed * BulletDir
+		BulletPower = 20
+
+	else:
+		assert(false) #NOTE(hugo) : Invalid default case
+
+	BulletNode.Velocity = BulletVelocity
+	BulletNode.Power = BulletPower
 	GameNode.add_child(BulletNode)
 	BulletNode.add_to_group(BulletsGroupName)
 
 func enable_shooting():
 	CanShoot = true
 
 func set_hit_side_to_left(EnteredHitbox):
-	if(!EnteredHitbox.is_in_group(BulletsGroupName)):
-		EnteredHitbox.queue_free()
-		if(!Frozen):
-			Frozen = true
-			FreezeTimerNode.start()
-			LastHitSide = "left"
-			emit_signal("hit_by_hitbox", self.get_name(), 10)
+	process_hit(EnteredHitbox, "left")
 
 func set_hit_side_to_right(EnteredHitbox):
+	process_hit(EnteredHitbox, "right")
```

Comment:
Naming : given what they do, `set_hit_side_to_right(EnteredHitbox)` and `process_hit(EnteredHitbox, "right")` should have a much more similar naming.
I've thought a few minutes about what that name could be, by looking at what the function does ; i suggest : 
`start_hitlag_from_right(EnteredHitbox)` and `start_hitlag(EnteredHitbox, "right")`